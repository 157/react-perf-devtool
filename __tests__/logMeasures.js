var MEASURES = require('../samples/parsed')

function logMeasures(measures) {
  measures.forEach(
    ({
      componentName,
      mount,
      render,
      update,
      unmount,
      totalTimeSpent,
      percentTimeSpent,
      numberOfInstances,
      componentWillMount,
      componentDidMount,
      componentWillReceiveProps,
      shouldComponentUpdate,
      componentWillUpdate,
      componentDidUpdate,
      componentWillUnmount
    }) => {
      // The time is in millisecond (ms)
      // TODO: The data generated is generalized. Make it concrete!
      const data = {
        component: componentName,
        mount,
        render,
        update,
        unmount,
        totalTimeSpent,
        percentTimeSpent,
        numberOfInstances,
        componentWillMount,
        componentDidMount,
        componentWillReceiveProps,
        shouldComponentUpdate,
        componentWillUpdate,
        componentDidUpdate,
        componentWillUnmount
      }

      // Proxy
      console.log(data)
    }
  )
}

function logToConsole({ port, components }, measures) {
  if (!components) {
    logMeasures(measures)
  } else if (typeof components !== undefined && Array.isArray(components)) {
    var stats = getRequiredMeasures(components, measures)

    logMeasures(stats)
  }
}

function getRequiredMeasures(components, measures) {
  var requiredMeasures = []

  if (!Array.isArray(components)) {
    components = [components]
  }

  measures.forEach(measure => {
    if (components.includes(measure.componentName)) {
      requiredMeasures.push(measure)
    }
  })

  return requiredMeasures
}

describe('Log measures to console', () => {
  var options = {
    port: 3000,
    components: ['App']
  }

  it('logs stats for all the components', () => {
    var stats = logToConsole(options, MEASURES)
    stats
    expect(stats).toMatchSnapshot()
  })
})
