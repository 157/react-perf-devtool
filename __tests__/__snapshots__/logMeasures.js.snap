// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log measures to console logs stats for all the components 1`] = `undefined`;
